LoopKit/InsulinKit/DoseStore.swift

```diff
@@ -868,14 +868,30 @@ extension DoseStore {
 
     /// Attempts to store doses from pump events to insulin delivery store
     private func syncPumpEventsToInsulinDeliveryStore(after start: Date? = nil, resolveMutable: Bool = false, completion: @escaping (_ error: Error?) -> Void) {
-        insulinDeliveryStore.getLastImmutableBasalEndDate { (result) in
-            switch result {
-            case .success(let date):
+        insulinDeliveryStore.getLastImmutableBasalEndDate { (date) in
+            if let date {
+                var start = date
                 // Limit the query behavior to 24 hours
-                let date = max(date, self.recentStartDate)
-                self.savePumpEventsToInsulinDeliveryStore(after: start ?? date, resolveMutable: resolveMutable, completion: completion)
-            case .failure(let error):
-                completion(error)
+                start = max(start, self.recentStartDate)
+                self.savePumpEventsToInsulinDeliveryStore(after: start, resolveMutable: resolveMutable, completion: completion)
+            } else {
+                self.persistenceController.managedObjectContext.perform {
+                    do {
+                        let events = try self.getPumpEventObjects(chronological: true, limit: 1)
+                        if let firstPumpEvent = events.first {
+                            var start = firstPumpEvent.startDate
+                            // Limit the query behavior to 24 hours
+                            start = max(start, self.recentStartDate)
+                            self.savePumpEventsToInsulinDeliveryStore(after: start, resolveMutable: resolveMutable, completion: completion)
+                        } else {
+                            // No previous basal, and no pump events; nothing to store or infer
+                            completion(nil)
+                            return
+                        }
+                    } catch {
+
+                    }
```

Comment:
was this empty `catch` intentional? If so, maybe add a comment?

Reply:
Ah, good "catch". ;) It was missing the call, thanks.